static/app/components/dropdownMenu/item.tsx

```diff
@@ -203,38 +180,48 @@ function DropdownMenuItem({
         onClose?.();
         rootOverlayState?.close();
       },
-      closeOnSelect: to || externalHref ? false : closeOnSelect,
+      closeOnSelect: isLink ? false : closeOnSelect,
       isDisabled,
     },
     state,
-    listElementRef
+    innerWrapRef
   );
 
-  // Merged menu item props, class names are combined, event handlers chained,
-  // etc. See: https://react-spectrum.adobe.com/react-aria/mergeProps.html
-  const mergedProps = mergeProps(props, menuItemProps, hoverProps, keyboardProps);
-  const itemLabel = node.rendered ?? label;
   const makeInnerWrapProps = () => {
     if (to) {
-      return {as: Link, to};
+      return {
+        as: Link,
+        to,
+      };
     }
 
     if (externalHref) {
-      return {as: ExternalLink, href: externalHref};
+      return {
+        as: ExternalLink,
+        href: externalHref,
+      };
     }
 
     return {as: 'div' as const};
   };
+  const mergedMenuItemContentProps = mergeProps(
+    props,
+    menuItemProps,
+    hoverProps,
+    keyboardProps,
+    makeInnerWrapProps(),
+    {ref: innerWrapRef, 'data-test-id': key}
```

Comment:
why is that `data-test-id` needed? As far as I can see, all tests use a role-based selector (which is the preferred solution anyway)

Reply:
Unfortunately there are quite a few tests using the dropdown component which do rely on the data-test-id 😞 

I agree that using roles is better (and the vast majority do that), but I didn't want to convert those tests in this PR